### PR TITLE
Dashboard for analysis reports

### PR DESCRIPTION
**Pull request by MaximilianWagner**
Add Report Dashboard, able to:
- display finished analysis
- download emba logs of finished emba analysis

Changes:
- extended Firmware model
- boundedExecutor is now class based

New:
- Tests for Firmware model
- DownloadFirmware model + form
- download_zipped request

### CODE REVIEW

File: embark/uploader/boundedExecutor.py [hunk 1]
```diff
@@ -50,32 +57,59 @@ def run_shell_cmd(self, cmd):
...
+
+            # success
+            logger.info(f"Successful cleaned up: {cmd}")
 
         except Exception as ex:
             logger.error(f"{ex}")
-            # TODO: inform asgi to propagate error to frontend
 
-    """
+            # take care of cleanup
```
Comment (diegiesskanne) [inline embark/uploader/boundedExecutor.py:78]: a little bit of duplicated code in regards to the successful case above. maybe refactor later

File: embark/uploader/boundedExecutor.py [hunk 2]
```diff
@@ -89,44 +123,45 @@ def submit_firmware(self, firmware_flags, firmware_file):
...
 
-    """
+    @classmethod
+    def submit(cls, fn, *args, **kwargs):
+        """
         same as concurrent.futures.Executor#submit, but with queue
 
         params: see concurrent.futures.Executor#submit
 
         return: future on success, None on full queue
```
Comment (diegiesskanne) [inline embark/uploader/boundedExecutor.py:144]: what does the future contain?

File: embark/uploader/test_boundedExecutor.py [hunk 3]
```diff
@@ -1,26 +1,26 @@
 from django.test import TestCase
 
-from .boundedExecutor import BoundedExecutor
+from .boundedExecutor import BoundedExecutor, max_workers, max_queue
 
 
 class test_boundedExecutor(TestCase):
 
     def setUp(self):
```
Comment (diegiesskanne) [inline embark/uploader/test_boundedExecutor.py:8]: remove if it does nothing or add a TODO

File: embark/uploader/urls.py [hunk 4]
```diff
@@ -9,7 +9,9 @@
     path('home/upload/', views.upload_file, name='embark-File'),
     path('home/upload/save_file', views.save_file, name='embark-FileSave'),
     path('home/serviceDashboard/', views.serviceDashboard, name='embark-ServiceDashboard'),
+    path('home/reportDashboard/', views.reportDashboard, name='embark-ReportDashboard'),
     # debug
     path('progress/', views.progress, name='embark-progress'),
```
Comment (diegiesskanne) [inline embark/uploader/urls.py:14]: can be removed as well

**Review by diegiesskanne — APPROVED** [review]
good stuff

File: embark/uploader/boundedExecutor.py (hunk 2 above)
Comment (MaximilianWagner) [reply embark/uploader/boundedExecutor.py:144]: emba process future

File: embark/uploader/test_boundedExecutor.py (hunk 3 above)
Comment (MaximilianWagner) [reply embark/uploader/test_boundedExecutor.py:8]: For readability purpose.

File: embark/uploader/boundedExecutor.py (hunk 1 above)
Comment (MaximilianWagner) [reply embark/uploader/boundedExecutor.py:87]: Done

**Pull request merged**